feat: adds landing page scaffolding

Eein opened the pull request:
## What

Adds the landing page scaffolding for games list and the side about panel.

Have a couple of common test cases to verify rendering doesn't look weird.

These components can likely be more atomic, but would make more sense when styling.

Since intitial commit:
- added getLeaderboards to apiClient (this does not include the use of queries yet)
- cleans up some responsive design bits
- uses key for v-loop
- Removed: adds leaderboardgg colors to extend in `tailwind.config.js` - this is currently set to override, as the design team may want something else down the road. they are named very specifically as to not override the existing colors
- update: removes unnecessary interaction for scaffolding

Remaining Work
- [x] Use actual links for the leaderboards
- [x] temporary styling to make it more cohesive
- [x] Mock backend tests

## Link to Issue

Tangentially related to https://github.com/leaderboardsgg/leaderboard-site/issues/639

## Acceptance

Simple "should render" tests are provided

### Steps for testing

Should render.

### Images
![image](https://github.com/user-attachments/assets/c761fef5-7f0d-4a24-9921-b34e74fc0da5)

TheTedder reviewed (commented):
I appreciate the Leaderboard names lighting up on hover but it would be great if there was some way to tell that they were clickable links without having to hover over them. Maybe style them with an underline?

buffet-time commented on components/blocks/Landing/Landing.vue:
```diff
@@ -0,0 +1,23 @@
+      <LandingLeaderboards
+        v-if="!leaderboards.loading && leaderboards.data != null"
+        :leaderboards="leaderboards"
+      ></LandingLeaderboards>
```
nit: you can use a `self-closing` style here.

buffet-time reviewed (changes requested):
just 1 change

TheTedder commented on components/blocks/Landing/LandingLeaderboards.vue:
```diff
@@ -0,0 +1,34 @@
+const localePath = useLocalePath()
+
+interface LandingLeaderboardsProps {
+  leaderboards: ApiResponse<LeaderboardViewModel[]>
```
Using this ApiResponse class as the property type feels a bit weird, especially since the component only uses the data property. Consider simply taking a list of LBs as props instead.

TheTedder commented on components/blocks/Landing/Landing.test.ts:
```diff
@@ -0,0 +1,20 @@
+import Landing from './Landing.vue'
+
+describe('Landing Component', () => {
+  const games = [
+    { name: 'Getting Over It With Bennet Foddy' },
+    { name: 'Getting Over It With Bennet Foddy' },
+    { name: 'Getting Over It With Bennet Foddy' },
+  ]
+
+  it('should render without crashing', async () => {
+    const wrapper = await mountSuspended(Landing, {
+      props: {
+        games,
+      },
```
These props aren't actually used it seems. I think this will also actually call the backend because you haven't mocked getLeaderboards here.

TheTedder commented on components/blocks/Landing/LandingLeaderboards.test.ts:
```diff
@@ -0,0 +1,31 @@
+import Landing from './LandingLeaderboards.vue'
+import { useGetLeaderboards } from '~/composables/api'
+
+const mockSuccessGetLeaderboards = vi.fn(() => Promise.resolve({ ok: true }))
+vi.mock('lib/api/Leaderboards', () => ({
+  Leaderboards: function Leaderboards() {
+    this.getLeaderboards = mockSuccessGetLeaderboards
+  },
+}))
+
+const leaderboards = await useGetLeaderboards()
```
I don't think it makes sense to mock this import and then immediately call it. Just pass the test data into the component directly.

Eein commented:
Rebased, and made the changes requested - 

Removed the mocks completely since this isn't doing any integration tests - more or less copy pasted from elsewhere which is why it was the way it is.
